controllers/grid/users/chapter/form/ChapterForm.inc.php

```diff
@@ -157,31 +160,33 @@ function fetch($request, $template = null, $display = false) {
 			}
 		}
 
-		$templateMgr = TemplateManager::getManager($request);
 		$templateMgr->assign([
 			'chapterAuthorOptions' => $chapterAuthorOptions,
 			'selectedChapterAuthors' => array_map(function($author) { return $author->getId(); }, $selectedChapterAuthors),
 		]);
 
 		if ($this->getChapter()) {
 			$submissionFiles = Services::get('submissionFile')->getMany(['submissionIds' => [$this->getMonograph()->getId()]]);
-			$chapterFileOptions = [];
-			$selectedChapterFiles = [];
 			foreach ($submissionFiles as $submissionFile) {
-				if (!$submissionFile->getData('chapterId') || $submissionFile->getData('chapterId') == $this->getChapter()->getId()) {
-					$chapterFileOptions[$submissionFile->getId()] = $submissionFile->getLocalizedData('name');
-				}
 				if ($submissionFile->getData('chapterId') == $this->getChapter()->getId()) {
 					$selectedChapterFiles[] = $submissionFile->getId();
 				}
 			}
-			$templateMgr = TemplateManager::getManager($request);
-			$templateMgr->assign([
-				'chapterFileOptions' => $chapterFileOptions,
-				'selectedChapterFiles' => $selectedChapterFiles,
-			]);
 		}
 
+		$submissionFiles = Services::get('submissionFile')->getMany(['submissionIds' => [$this->getMonograph()->getId()]]);
```

Comment:
It looks like this duplicates a call to get the files at line 169. Can you get the submission files once and loop over them once to get the `chapterFileOptions` and the `selectedChapterFiles`?

Reply:
Yeah sure, probably that is possible. That seemed like a clean way of doing it, but you are right that it causes unnecessary queries which we already have plenty.